src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -76,90 +75,97 @@ private static bool GetConditionValue(string conditionName, Func<bool> detectFun
+            return GetConditionValue(nameof(Is_Windows),
+                                     ConditionalTestDetectors.IsWindows);
         }
 
         public static bool Domain_Joined()
         {
-            // Temporarily use the simple heuristic that if we are running the services locally, it is.
-            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
             return GetConditionValue(nameof(Domain_Joined),
-                                     Is_Server_Local);
+                                     () => ConditionalTestDetectors.IsClientDomainJoined() && 
+                                           ConditionalTestDetectors.IsServerDomainJoined());
         }
 
         public static bool Root_Certificate_Installed()
         {
-            // Temporarily use the simple heuristic that if we are running the services locally, it is.
-            // Refactor this to test only when solving https://github.com/dotnet/wcf/issues/1024 
             return GetConditionValue(nameof(Root_Certificate_Installed),
-                                     Is_Server_Local);
+                                     ConditionalTestDetectors.IsRootCertificateInstalled);
         }
 
         public static bool Client_Certificate_Installed()
         {
-            // Temporarily use the simple heuristic that if we are running the services locally, it is.
-            // Refactor this to test only when solving https://github.com/dotnet/wcf/issues/1024 
             return GetConditionValue(nameof(Client_Certificate_Installed),
-                                     Is_Server_Local);
+                                     ConditionalTestDetectors.IsClientCertificateInstalled);
+        }
+
+        public static bool UserName_And_Password_Available()
+        {
+            return GetConditionValue(nameof(UserName_And_Password_Available),
+                                     ConditionalTestDetectors.AreUserNameAndPasswordAvailable);
         }
 
         public static bool SPN_Available()
         {
             // Temporarily use the simple heuristic that if we are running the services locally, it is.
             // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
             return GetConditionValue(nameof(SPN_Available),
-                                     Is_Server_Local);
+                                     Server_Is_LocalHost);
         }
 
         public static bool Kerberos_Available()
         {
             // Temporarily use the simple heuristic that if we are running the services locally, it is.
             // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
             return GetConditionValue(nameof(Kerberos_Available),
-                                     Is_Server_Local);
+                                     Server_Is_LocalHost);
+        }
+
+        public static bool Server_Accepts_Certificates()
+        {
```

Comment:
Reviewers: as we discussed @hongdai will investigate and provide the complete list for conditions like Basic, Digest, website-allows-certificates, etc.  For now I added the bare minimum I knew so we could mark the tests that currently fail in CI running in the lab.  I will add the others or rename these in later PR's.